core/tee/tee_svc.c

```diff
@@ -828,10 +841,14 @@ static TEE_Result tee_svc_update_out_param(
 			 * size needs to be updated.
 			 */
 			sz = param->u[n].mem.size;
-			if (tmp_buf_va[n] && sz <= vals[n * 2 + 1]) {
+
+			res = GET_USER_SCALAR(in_sz, &vals[n * 2 + 1]);
+			if (res)
+				return res;
+
+			if (tmp_buf_va[n] && sz <= in_sz) {
 				void *src = tmp_buf_va[n];
 				void *dst = (void *)(uintptr_t)vals[n * 2];
```

Comment:
`GET_USER_SCALAR()` missing for `&vals[n * 2]`

Reply:
Thanks for the catch. updated